nomad/server.go

```diff
@@ -1088,7 +1088,10 @@ func (s *Server) setupVaultClient() error {
 // setupRPC is used to setup the RPC listener
 func (s *Server) setupRPC(tlsWrap tlsutil.RegionWrapper) error {
 	// Populate the static RPC server
-	s.setupRpcServer(s.rpcServer, nil)
+	err := s.setupRpcServer(s.rpcServer, nil)
+	if err != nil {
+		return err
+	}
```

Comment:
Note for reviewers: I'm not wild about how this is getting wired up right now because we have to do disk IO in the middle of setting up the RPC server. But I'm also 90% certain it's going to end up being changed when I implement the key replication goroutine, so rather than implementing it twice I've left this in a working but not quite ideal state.